aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -227,12 +227,18 @@ static void bitwise_not_kernel(TensorIterator& iter) {
 }
 
 static void frac_kernel(TensorIterator& iter) {
-  AT_DISPATCH_FLOATING_TYPES_AND2(kBFloat16, kHalf, iter.dtype(), "frac_cpu", [&]() {
-    cpu_kernel_vec(
-        iter,
-        [=](scalar_t a) -> scalar_t { return a - std::trunc(a); },
-        [=](Vec256<scalar_t> a) { return a.frac(); });
-  });
+  AT_DISPATCH_FLOATING_TYPES_AND2(
+      kBFloat16, kHalf, iter.dtype(), "frac_cpu", [&]() {
+        cpu_kernel_vec(
+            iter,
+            [=](scalar_t a) -> scalar_t {
+              if (std::isinf(a)) {
```

Comment:
Could this use std::modf instead?

Reply:
I'll benchmark and update if `modf` is more efficient.
Thanks!

Reply:
Since `c10::Half` and `BFloat16` don't support std::modf, it will be trickier. Also note that we'll be updating only the non-vectorized part (which won't gain us a lot of perf)